lib/pmp/pmp.go

```diff
@@ -66,10 +67,8 @@ func Discover(ctx context.Context, renewal, timeout time.Duration) []nat.Device
 	conn, err := (&net.Dialer{}).DialContext(timeoutCtx, "udp", net.JoinHostPort(ip.String(), "5351"))
 	if err == nil {
 		conn.Close()
-		localIPAddress, _, err := net.SplitHostPort(conn.LocalAddr().String())
-		if err == nil {
-			localIP = net.ParseIP(localIPAddress)
-		} else {
+		localIP, _ = osutil.IPFromAddr(conn.LocalAddr())
+		if localIP == nil {
```

Comment:
err is nil in the line below, so some tweak required here to the error handling

Reply:
This function doesn't return an error.

Reply:
Is that why DeepSource is failing? It's giving me a 500.

Reply:
It may not return an error, but we're logging an error which is guaranteed to be nil

Reply:
Oh, right! Fixed.